tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +214,97 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	type watcher struct {
+		prefix string
+		c      *clientv3.Client
+		wc     clientv3.WatchChan
+	}
+	prefixes := 36
+	watchPerKey := 10
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string)
+	ws := make([]watcher, prefixes*watchPerKey)
+
+	var (
+		wg  sync.WaitGroup
+		pr  *clientv3.PutResponse
+		err error
+	)
+
+	const (
+		chars = "abcdefghijklmnopqrstuvwxyz0123456789"
+	)
+
+	wg.Add(len(ws))
+	for i := 0; i < len(ws); {
+		for _, prefix := range chars {
+			go func(w *watcher, i int, prefix rune) {
```

Comment:
@xiang90 when i had put the logs it was taking approx 1s per watch, and since there are 360 watches being setup on a single cluster its taking close to 360s. Do you want me to dig into it for the reason? What do you feel approx time it should take for a single watch?